### ISSUE TITLE
compilation error on mac

### ISSUE THREAD

**Issue by jamesalee213** [akermu/emacs-libvterm#389]
I was getting compilation error when trying to compile libvterm on my machine:

macOS Catalina version 10.15.6
had installed cmake and libtool with brew before compilation

![image](https://user-images.githubusercontent.com/20655571/91326394-25d50d00-e779-11ea-9fb5-d8d0454f2675.png)

I don't know if there's a better way to solve this, but my workaround solution was to add 
`#define _POSIX_C_SOURCE 200809L`
on top of `vterm-ctrl.c` file
I got this solution from https://stackoverflow.com/questions/26284110/strdup-confused-about-warnings-implicit-declaration-makes-pointer-with

I was wondering if any other people had similar issues trying to compile on mac and if there was a better solution. Also, was wondering if the source code can be updated.

**Comment by Sbozzolo** [comment]
It seems that this has just been fixed upstream: https://github.com/neovim/libvterm/commit/54c03b21f763fa775a4c0643a9d8326342873179

Can you please check that the solution implemented there works?

**Comment by Sbozzolo** [comment]
You can check out the latest version of libvterm changing the commit has in CMakelists.txt (line 61) to
54c03b21f763fa775a4c0643a9d8326342873179

Then, recompile.

**Comment by jamesalee213** [comment]
Changing the commit didn't seem to have worked. 
I get another compilation error 
```
 error: passing 'VTermStringFragment' to parameter of incompatible type 'char *'
    term_set_title(term, val->string);
```

**Comment by Sbozzolo** [comment]
Thanks for testing this out and providing a workaround. This is most likely an issue with libvterm, and not with our package. Please, report it to the maintainers of libvterm.

**Comment by Sbozzolo** [comment]
I just tried to compile it myself. 

> Changing the commit didn't seem to have worked.
> I get another compilation error
> 
> ```
>  error: passing 'VTermStringFragment' to parameter of incompatible type 'char *'
>     term_set_title(term, val->string);
> ```

This is actually our bug. I'll fix it.

**Comment by cedarbaum** [comment]
As another point of data, I was able to work around this by changing the first line in`libvterm/bin/vterm-ctrl.c`:

```diff
- #define _XOPEN_SOURCE 500  /* strdup */
+ #define _XOPEN_SOURCE 600  /* strdup */
```

**Comment by Sbozzolo** [comment]
> As another point of data, I was able to work around this by changing the first line in`libvterm/bin/vterm-ctrl.c`:
> 
> ```diff
> - #define _XOPEN_SOURCE 500  /* strdup */
> + #define _XOPEN_SOURCE 600  /* strdup */
> ```

Thanks. That was indeed https://github.com/neovim/libvterm/commit/54c03b21f763fa775a4c0643a9d8326342873179.

Unfortunately, `emacs-libvterm` is currently incompatible with the latest releases of `libvterm`. It will be take some work to update our code maintaining backwards compability.

**Comment by kgrimes2** [comment]
@Sbozzolo Can we pinpoint the `libvterm` release that breaks `emacs-libvterm`? It might be worth updating the README, as it currently only recommends v0.1 or later.

**Comment by Sbozzolo** [comment]
The problem is the new struct `VTermStringFragment`. Versions up to 0.1.3 will work fine (I believed this was the latest released, but it looks like there's a 0.1.4 around).
